### DISCUSSION TITLE
How to securely get access to dmmf.datamodel?

### DISCUSSION THREAD

**Discussion by berezhinskiyo** [prisma/prisma#23749] [Q&A]
### Question

Hi, I've made the prisma client extension and some internal logic needed to have access to dmmf.datamodel (metadata of dmmf.datamodel). I got it from `import { Prisma  } from '@prisma/client'; 
Prisma.dmmf.datamodel.models`, but sometimes when I load new version of prisma I got following error `TypeError: Cannot read properties of undefined (reading 'datamodel')` because Prisma.dmmf === null = true. How to get dmmf correctly, securely? 

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @berezhinskiyo 👋 

Can you please show the code for how you are using the `Prisma.dmmf` with the Client extension please?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
